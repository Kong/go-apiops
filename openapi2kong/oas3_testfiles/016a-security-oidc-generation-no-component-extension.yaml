# security directives can generate oidc plugin configs

openapi: '3.0.0'
info:
  title: OIDC tests
  version: v2
servers:
  - url: /

# defining a toplevel security spec
security:
  - doclevel-oidc: ["top-scope"]

# overriding the top level with per operation ones below
paths:
  /path1:
    get:
      operationId: OpsId1
      # this should get the plugin per doclevel-oidc specs, since we do not override
      summary: List API versions
      responses:
        '200':
          description: |-
            200 response


components:
  securitySchemes:
    doclevel-oidc:
      type: openIdConnect
      openIdConnectUrl: https://konghq.com/top-level
      x-kong-security-openid-connect:
        config:
          run_on_preflight: false
          scopes_required: ["scope2", "scope1"]