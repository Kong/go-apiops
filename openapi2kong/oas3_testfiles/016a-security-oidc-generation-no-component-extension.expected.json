{
  "_format_version": "3.0",
  "services": [
    {
      "host": "localhost",
      "id": "bcda787b-18bc-5f83-85f2-1cf83b5f8eb3",
      "name": "oidc-tests",
      "path": "/",
      "plugins": [
        {
          "config": {
            "issuer": "https://konghq.com/top-level",
            "run_on_preflight": false,
            "scopes_required": [
              "scope1",
              "scope2",
              "top-scope"
            ]
          },
          "name": "openid-connect"
        }
      ],
      "port": 443,
      "protocol": "https",
      "routes": [
        {
          "id": "558eb5de-46b9-5500-ae2a-9a8e321e9d65",
          "methods": [
            "GET"
          ],
          "name": "oidc-tests_opsid1",
          "paths": [
            "~/path1$"
          ],
          "plugins": [],
          "regex_priority": 200,
          "strip_path": false,
          "tags": [
            "OAS3_import",
            "OAS3file_016a-security-oidc-generation-no-component-extension.yaml"
          ]
        }
      ],
      "tags": [
        "OAS3_import",
        "OAS3file_016a-security-oidc-generation-no-component-extension.yaml"
      ]
    }
  ],
  "upstreams": []
}